Code/C++/Luhns_algorithm.cpp

```diff
@@ -0,0 +1,97 @@
+
+Checksum = 1 + 9 + 7 + 7 + 3 + 5 + 8 + 0 + 4 + 4 + 3 + 8 + 2 + 3 + 1 + 5 = 70
+
+Since the checksum is divisible by ten, 5987654024681355 is a valid credit card number. Write a program
+that, upon input a 16-digit credit card number, determines whether the card number is valid. A 16-digit
+integer cannot be accommodated in a long int variable, so read the input as a string (a character array).
+************************************************************************************/
+
+#include <bits/stdc++.h>
+using namespace std;
+
+//global var
+int checksum = 0;
+
+//function to evaluate checksum according to Luhn's Algorithm
+void Luhn_checksum_calc(char card[])
+{
+    //iterates as many times as there are characters in the input string
+    for (int i = 0; i < strlen(card); i++)
+    {
+        int digit = card[i] - '0';
+        //for even positions
+        if (i%2==1){
+            //digits added to checksum as it is (unchanged)
+            checksum += digit;
+            cout << digit;
+        }
+        //for odd positions (will require 2 digit adjustments if 2*digit>9)
+        if (i%2==0){
+            checksum += (2*digit<10) ? (2*digit):((2*digit)%10)+1;
+            /*  Ternary operator used here to imply single digits are added to checksum without any change;
+            But for 2digit nos., sum of the digits is to be added to checksum.
+            (n modulo 10) + 1 yields same value as the 2 digit adjustment
+            (eg, (12%10)+1 = 3 = 1+2   ;   (10%10)+1 =1 = 1+0     ; (18%10)+1 = 9 = 1+8 )   */
+            cout << (2*digit<10) ? (2*digit):(((2*digit)%10)+1);
+        }
+    }
+}
+
+//main driver code
+int main()
+{
+    //declare character array
+    char card[16];
+    cout << "Enter credit card number:" << endl;
+    //reads a line from input and stores it into the 'card' character array
+    gets (card);
+    //function to evaluate checksum
+    Luhn_checksum_calc(card);
+    //test if checksum is valid (ie, if modulo10 = 0)
+    cout <<"\t =  " << checksum;
+    cout << "\t" << ((int)strlen(card) == 16 && checksum%10==0 ? "(Valid)":"(Invalid)") << endl;       
+    //the card number must have exactly 16 digits   
+    if ((int)strlen(card) != 16)
+        cout << "[Invalid because card number must have exactly 16 digits.]" << endl;            
+    return 0; 
+}
+
+/*
+Time Complexity = O(1)
+Space Complexity = O(1)
+
+Std O/P 1:
+Enter the credit card number:
+5987654024681355        
+ + 1 + 9 + 7 + 7 + 3 + 5 + 8 + 0 + 4 + 4 + 3 + 8 + 2 + 3 + 1 + 5         = 70  (Valid)
+
+Std O/P 2:
+Enter the credit card number:
+5987654024681353
+ + 1 + 9 + 7 + 7 + 3 + 5 + 8 + 0 + 4 + 4 + 3 + 8 + 2 + 3 + 1 + 3         = 68  (Invalid)
+
+Std O/P 3:
+Enter the credit card number:
+59876540246813550
+ + 1 + 9 + 7 + 7 + 3 + 5 + 8 + 0 + 4 + 4 + 3 + 8 + 2 + 3 + 1 + 5 + 0     = 70  (Invalid)
+[Invalid because card number must have exactly 16 digits.]
```

Comment:
Write the output just as it appears on the console.

Reply:
Thats how it appears in the console...

Reply:
`+` is never printed by the code in between the numbers. Yet, it still appears in your I/O exampls.